src/lib/y2network/route.rb

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+module Y2Network
+  # This class represents a network route
+  class Route
+    # @return [IPAddr]
+    attr_reader :to
+    # @return [Interface]
+    attr_reader :interface
+    # @return [IPAddr,nil]
+    attr_reader :source
+    # @return [IPAddr,nil]
+    attr_reader :gateway
+
+    def initialize(to, interface, gateway: nil, source: nil, preference: nil)
```

Comment:
so for default route is has to look like `Route.new(nil, "eth0")`? And whats more how will look static route without interface? like `Route.new(nil, "-")`? (see https://bugzilla.suse.com/show_bug.cgi?id=93996 ) This does not look nice.
Can we at least have ability to pass ":default` as to and only internally keep it as nil and for interface having it optional or allowing to pass symbol `:any` ?

Reply:
Also to have easier review can you add explanation what means nil in gateway, source and preference, to see if it makes sense? Thanks

Reply:
That's open for discussion: we could do the `to` parameter optional. And we could even use the same approach for static routes without and interface. Please, beware that we are not passing strings as interfaces anymore (we have a proper `Y2Network::Interface` class for that.

As a side note, I would say that, at low level, all routes have a device. But it is true that in the `sysconfig` files you can leave them out.